nodes/test/test_ssrf_protection.py

```
@@ -0,0 +1,304 @@
+
+def _fake_getaddrinfo(ip: str):
+    """Return a patched getaddrinfo that always resolves to *ip*."""
+
+    def _patched(host, port, **_kw):
+        family = socket.AF_INET6 if ':' in ip else socket.AF_INET
+        return [(family, socket.SOCK_STREAM, socket.IPPROTO_TCP, '', (ip, port))]
+
+    return _patched
+
+
+def _fake_getaddrinfo_multi(ips: List[Tuple[str, int]]):
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `_fake_getaddrinfo_multi`

(ANN202)